assets/js/settings.js

```diff
@@ -2,6 +2,39 @@
 	function ( $, window, undefined ) {
 		'use strict';
 
+    // storing RegExp strings for decoding the uploaded config file
+    var RegExpStrings = {
+      channel_id: /channel_id: [0-9a-zA-Z_-]*/g,
+      key: /key: [0-9a-zA-Z_-]*/g,
+      secret: /secret: [0-9a-zA-Z_-]*/g
+    }
+
+    function updateCreds(input) {
+      $( '#api_channel' ).val(
+        input.match(RegExpStrings.channel_id).toString().split(" ")[1]
```

Comment:
If there is no match, this is going to fail. You can simplify the check by putting the matcher in the regex, like this:

```javascript
/channel_id: ([0-9a-zA-Z_-]+)/
```

And then you can change this to a two-step process, like this:

```javascript
var channelIdMatch = input.match(RegExpStrings.channel_id);
$( '#api_channel' ).val(channelIdMatch ? channelIdMatch[1] : '');
// etc
```